### ISSUE TITLE
in ActiveRecod parent::find() behaves exactly like static::find()

### ISSUE THREAD

**Issue by Faryshta** [yiisoft/yii2#8870]
specially annoying when i am trying to do extend the active record for example for a Search class

**Comment by cebe** [comment]
whats the problem? please give more details, ideally some code.

**Comment by Faryshta** [comment]
```php
class CategoriaSearch extends Categoria
{
    /**
     * Creates data provider instance with search query applied
     *
     * @param array $params
     * @return ActiveDataProvider
     */
    public function search($params)
    {
        $query = parent::find();
        $dataProvider=new ActiveDataProvider([
            'query'=> $query,
        ]);
        // rest of the code Active provider code
    }
}
```

The dataprovider generated here, will return an array for [CategorySearch] models including the rewritten methods i have only for search (like `rules()`) avoiding me to properly use things i need to behave differently on the search than on the create/update (like using date ranges on the search)

**Comment by cebe** [comment]
this is because of late static binding, the method gets called in the context of the search model. Have you tried Categoria::find() ? not sure but it might make a difference.

**Comment by Faryshta** [comment]
Indeed Category::find() works perfectly, the issue is that i have many similar models and searches so i want to make a trait to generate the query with default actions (example date range in all the searches) and i need to start the query with `parent::find()` so that it works with all the models i need.

**Comment by Faryshta** [comment]
also `parent::className()` returns the same as `static::className()` you can test it right now.

**Comment by cebe** [comment]
hm... http://3v4l.org/KsD35 not sure if that is a php bug. I would expect parent to behave differently. Not sure if there is a good reason for current behavior.

**Comment by Faryshta** [comment]
all the links i can find seem to imply that the only way to fix this is use `get_parent_class` instead of `parent::className()` which imo would be redundant to have two ways to determine the classname of a class or the `::class` constant in php 5.5.

**Comment by Faryshta** [comment]
@cebe i think this should be closed since it will get solved by 2.1
